### ISSUE TITLE
[feature] Add DeepSeek-V3, Llama 4 and Qwen3-MoE model architectures

### ISSUE THREAD

**Issue by astorise** [huggingface/candle#3653]
### Motivation
`candle-transformers` already covers a lot (llama, qwen2/qwen3, deepseek2,
gemma2/3/4, glm4, mixtral, plus VLMs), but a few recent dense/MoE families that
users frequently ask for are still missing.

### Gap in candle (please confirm)
- **DeepSeek-V3** — only `deepseek2` (V2) is present; V3 adds aux-loss-free MoE
  routing, Multi-Token Prediction and FP8.
- **Llama 4** — MoE + interleaved/iRoPE; not present.
- **Qwen3-MoE** — only dense `qwen3` appears to be present.

### Proposed implementation
- New modules `candle-transformers/src/models/{deepseek3,llama4,qwen3_moe}.rs`,
  following the existing MoE models (`mixtral`, `deepseek2`).
- HF `config.json` parsing + `VarBuilder` loading + quantized variant where it
  makes sense.
- A minimal `candle-examples/examples/<model>` per family for validation.

### Use-case / example
```rust
// Load Qwen3-MoE like any other candle-transformers model.
let cfg: Qwen3MoeConfig = serde_json::from_slice(&std::fs::read("config.json")?)?;
let model = Qwen3Moe::load(vb, &cfg)?;
let logits = model.forward(&input_ids, /*pos=*/0)?;
```

**Comment by haydonryan** [comment]
Deepseek v4 flash is my target model on dual rtx6000 pro. Currently candle doesn't support this either. 
Interestingly vllm does not support it on Blackwell as of v0.24. 

It'd be great to support it before they did. I can put some cycles and tokens into it.

**Comment by astorise** [comment]
> Deepseek v4 flash is my target model on dual rtx6000 pro. Currently candle doesn't support this either. Interestingly vllm does not support it on Blackwell as of v0.24.
> 
> It'd be great to support it before they did. I can put some cycles and tokens into it.

Thanks for the interest — contributions are very welcome!

I'm not familiar yet with the exact DeepSeek V4 Flash architecture, so the first step would be comparing its config.json / modeling_deepseek*.py against DeepSeek-V3 to identify what's new or changed. The V3 implementation here ([candle-transformers/src/models/deepseek3.rs](https://github.com/astorise/candle/blob/claude/ci-arc-gpu-runner/candle-transformers/src/models/deepseek3.rs)) can serve as a starting point if the overall structure is similar (MLA attention, MoE routing…).

One practical tip from this PR: yujiepan on the Hub often publishes tiny randomly-initialized checkpoints (e.g. yujiepan/deepseek-v3-tiny-random) that let you develop and test the full loading + forward path without needing the real multi-hundred-GB weights. Worth checking if one exists for V4 Flash before you start, or creating one yourself with the same pattern.

Happy to review a WIP PR early if you want feedback on the architecture mapping.

**Comment by haydonryan** [comment]
Thank you for the suggestion. I was able to find: https://huggingface.co/yujiepan/deepseek-v4-bf16-tiny-random. I'll do some learning and hopefully make a start.
